sdk/azidentity/azure_pipelines_credential.go

```diff
@@ -98,27 +98,32 @@ func (a *AzurePipelinesCredential) getAssertion(ctx context.Context) (string, er
 	url := a.oidcURI + "?api-version=" + oidcAPIVersion + "&serviceConnectionId=" + a.connectionID
 	url, err := runtime.EncodeQueryParams(url)
 	if err != nil {
-		return "", err
+		return "", newAuthenticationFailedError(credNameAzurePipelines, "couldn't send OIDC token request: "+err.Error(), nil, nil)
```

Comment:
It might be good to slightly change the string "couldn't send OIDC token request" for each case to disambiguate which error was hit.